internal/fuzzbinder/fuzzclaim.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package fuzzbinder
+
+// This file provides a custom `ClaimSpec` type, FuzzClaimSpec, to be used
+// for fuzzing claims within the ClaimPredicate (defined in amber package).
+// FuzzClaimSpec is intended to be used for providing the user with the
+// needed elements to characterize the security of a revision of the source
+// code based on fuzzing.
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/project-oak/transparent-release/pkg/amber"
+)
+
+// FuzzClaimV1 is the URI that should be used as the ClaimType in V1 Amber
+// Claim representing a V1 Fuzz Claim.
+const FuzzClaimV1 = "https://github.com/project-oak/transparent-release/fuzz_claim/v1"
+
+// FuzzClaimSpec gives the `ClaimSpec` definition.
+type FuzzClaimSpec struct {
+	// `ClaimSpec` per fuzz-target.
+	PerTarget []PerTargetSpec `json:"perTarget"`
+	// `ClaimSpec` for all the fuzz-targets.
+	PerProject *PerProjectSpec `json:"perProject"`
+}
+
+// PerTargetSpec contains the fuzzing claims specification per fuzz-target.
+type PerTargetSpec struct {
+	// Name of the fuzz-target.
+	Name string `json:"name"`
+	// URI of the fuzz-target.
+	URI string `json:"uri"`
+	// Coverage specifies the code coverage by the fuzz-target.
+	Coverage *FuzzCoverage `json:"coverage"`
+	// Bugs specifies the number of detected bugs using the fuzz-target.
+	Bugs int `json:"bugs"`
+	// Crashes specifies the number of crashes obtained using the fuzz-target.
+	Crashes int `json:"crashes"`
+	// FuzzEffort specifies the fuzzing efforts spent while using the fuzz-target.
+	FuzzEffort *FuzzEffortSpec `json:"fuzzEffort,omitempty"`
+}
+
+// PerProjectSpec contains the fuzzing claims specification of the revision
+// of the source code for all fuzz-targets.
+type PerProjectSpec struct {
+	// Coverage specifies the code coverage by all fuzz-targets.
```

Comment:
Do we want to include the name/uri of the project, too? Similar to the PerTargetSpec.

Reply:
This is a good point. I think `PerProjectSpec` is supposed to be an aggregation over `PerTargetSpec` for all the fuzz-targets. So this should be implicit, unless somehow there are fuzz-targets that are considered here, but not included in the list of `PerTargetSpec`s. 

I am not sure if that scenario is possible. In any case, I think it would be nice to state this relationship between `PerTarger` and `PreProject` fields. Is there a formula that could be used for deriving the `PerProject` from `PerTarget` list?

Reply:
Yes, `PerProject`  should include all the fuzz targets in the GitHub repo. It's not possible to derive the `PerProject`  from `PerTarget`  without using the evidence or fuzz reports because the aggregation of coverage depends on the intersection of covered lines/branches by the fuzz-targets.